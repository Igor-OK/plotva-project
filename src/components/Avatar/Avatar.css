.avatar {
  display: inline-block;
  overflow: hidden;   
  border-radius: 100%;  
}

.avatar_large {
  flex: 0 0 62px;  
  width: 62px;
  height: 62px;
}

.avatar_medium {
  flex: 0 0 60px;  
  width: 60px;
  height: 60px;
}

.avatar_small {
  flex: 0 0 40px;  
  width: 40px;
  height: 40px;
}

.avatar__img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}