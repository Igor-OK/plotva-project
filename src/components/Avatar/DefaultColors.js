export const Colors = [
  '#e17076',
  '#7bc862',
  '#e5ca77',
  '#65aadd',
  '#a695e7',
  '#ee7aae',
  '#6ec9cb',
  '#faa774',
  '#e17076',
  '#7bc862',  
];
