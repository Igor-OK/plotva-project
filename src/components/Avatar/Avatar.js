import React, { Component } from 'react';
import './Avatar.css';
import defaultImage from './default.png'
export class Avatar extends Component {
  render() {
    let {size = 'large', img = defaultImage } = this.props;
    return (
      <div className={`avatar avatar_${size}`}>
        <img className="avatar__img" src={img} alt='avatar'></img>
      </div>
    );
  }
}
