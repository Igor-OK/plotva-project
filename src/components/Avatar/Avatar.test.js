import React from 'react';
import { Avatar } from './Avatar';
import renderer from 'react-test-renderer';

import { shallow, mount, render } from 'enzyme';

test('Avatar render', () => {
  const component = renderer.create(
    <Avatar size = 'large' defaultName = 'AB' />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

// test('Avatar without image should render default color circle', () => {
//     expect(shallow(<Avatar size = 'large' defaultName = 'AB' />)
//       .contains(<div className="avatar_default">AB</div>))
//         .toBe(true);
// });