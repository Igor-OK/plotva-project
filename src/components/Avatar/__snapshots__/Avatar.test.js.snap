// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar render 1`] = `
<div
  className="avatar avatar_large "
>
  <div
    className="avatar_default"
    style={
      Object {
        "backgroundColor": "#e17076",
      }
    }
  >
    AB
  </div>
</div>
`;
